test/e2e/framework/util.go

```diff
@@ -4576,6 +4576,122 @@ func CheckConnectivityToHost(f *Framework, nodeName, podName, host string, pingC
 	return err
 }
 
+// Regression test for #74839
+// https://github.com/kubernetes/kubernetes/issues/74839
+func CheckConnectionReset(f *Framework) error {
+	serverLabel := map[string]string{
+		"app": "boom-server",
+	}
+	clientLabel := map[string]string{
+		"app": "client",
+	}
+
+	serverPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "boom-server",
+			Labels: serverLabel,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				v1.Container{
+					Name:  "boom-server",
+					Image: "gcr.io/kubernetes-e2e-test-images/regression-issue-74839-amd64:1.0",
+					Ports: []v1.ContainerPort{
+						v1.ContainerPort{
+							ContainerPort: 9000, // Default port exposed by boom-server
+						},
+					},
+				},
+			},
+			Affinity: &v1.Affinity{
+				PodAntiAffinity: &v1.PodAntiAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+						v1.PodAffinityTerm{
+							LabelSelector: &metav1.LabelSelector{
+								MatchLabels: clientLabel,
+							},
+							TopologyKey: "kubernetes.io/hostname",
+						},
+					},
+				},
+			},
+		},
+	}
+	_, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(serverPod)
+	if err != nil {
+		return err
+	}
+
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "boom-server",
+		},
+		Spec: v1.ServiceSpec{
+			Selector: serverLabel,
+			Ports: []v1.ServicePort{
+				v1.ServicePort{
+					Protocol: v1.ProtocolTCP,
+					Port:     9000,
+					// TargetPort: 9000,
+				},
+			},
+		},
+	}
+	_, err = f.ClientSet.CoreV1().Services(f.Namespace.Name).Create(svc)
+	if err != nil {
+		return err
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "startup-script",
+			Labels: clientLabel,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "startup-script",
+					Image: "gcr.io/google-containers/startup-script:v1",
+					Command: []string{
+						"bash", "-c", "while true; do sleep 2; nc boom-server 9000& done",
+					},
+				},
+			},
+			Affinity: &v1.Affinity{
+				PodAntiAffinity: &v1.PodAntiAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+						v1.PodAffinityTerm{
+							LabelSelector: &metav1.LabelSelector{
+								MatchLabels: serverLabel,
+							},
+							TopologyKey: "kubernetes.io/hostname",
+						},
+					},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	_, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(pod)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < 20; i++ {
+		time.Sleep(1 * time.Second)
+		resultPod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(serverPod.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		if resultPod.Status.ContainerStatuses[0].LastTerminationState.Terminated != nil {
```

Comment:
Curious: when would this pod fails? What happen if this pod starts before server pod becomes ready?

Reply:
The server pod will crash when the bug is triggered, we don't check the client pod..

Reply:
Ah I see, thanks. Seems straightforward. Would probably be nice to print the logs from the pod in this case so we know it is the same connection reset error?

Reply:
We do. but if it's something else, we cannot allow it to pass either. but I think you are right, it's better for debugging.